src/Microsoft.DotNet.Tools.Builder/Program.cs

```diff
@@ -0,0 +1,171 @@
+
+                //compile dependencies
+                if (!CompileDependencies(context, configValue, outputPath, intermediateOutputPath, noHost))
+                {
+                    return false;
+                }
+                
+                //compile project
+                if (!CompileProject(context, configValue, outputPath, intermediateOutputPath, noHost, isNative,
+                    isCppMode, archValue, ilcArgsValue, ilcPathValue, appDepSdkPath, ilcSdkPathValue))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static bool CompileDependencies(ProjectContext context, string configuration, string outputPath, string intermediateOutputPath, bool noHost)
+        {
+            // Create the library exporter
+            var exporter = context.CreateExporter(configuration);
+
+            // Gather exports for the project
+            var dependencies = exporter.GetDependencies().ToList();
+
+            var projects = new Dictionary<string, ProjectDescription>();
+
+            // Build project references
+            foreach (var dependency in dependencies)
+            {
+                var projectDependency = dependency.Library as ProjectDescription;
+
+                if (projectDependency != null && projectDependency.Project.Files.SourceFiles.Any())
+                {
+                    projects[projectDependency.Identity.Name] = projectDependency;
+                }
+            }
+
+            foreach (var projectDependency in Sort(projects))
+            {
+                if (!CompileDependency(projectDependency, configuration, noHost, outputPath, intermediateOutputPath))
+                    return false;
+            }
+
+            projects.Clear()
+;
+            return true;
+        }
+
+        private static bool CompileDependency(ProjectDescription projectDependency, string configuration, bool noHost, string outputPath, string intermediateOutputPath)
+        {
+            var compileResult = Command.Create("dotnet-compile",
+                $"--framework {projectDependency.Framework} " +
+                $"--configuration {configuration} " +
+                $"--output \"{outputPath}\" " +
+                $"--temp-output \"{intermediateOutputPath}\" " +
+                (noHost ? "--no-host " : string.Empty) +
+                $"\"{projectDependency.Project.ProjectDirectory}\"")
+                .ForwardStdOut()
+                .ForwardStdErr()
+                .Execute();
+
+            return compileResult.ExitCode == 0;
+        }
+
+        private static bool CompileProject(ProjectContext context, string configValue, string outputPath,
+            string intermediateOutputPath, bool isNoHost, bool isNative, bool isCppMode, string archValue, string ilcArgsValue,
+            string ilcPathValue, string ilcSdkPathValue, string appDepSdkPath)
+        {
```

Comment:
@gkhanna79 @schellap 

Incremental compilation will be time stamp based and the granularity is an entire project
Before invoking the Compile verb, the Build verb computes all the input and output files for that project. If all output files are older than all input files, we skip the compilation of the project.

As discussed in the 2 design meetings we've had with the CLI team, for the January and February deadlines we'll do a shared library that both Compile and Build use to compute inputs and outputs. Post Jan / Feb we want to come up with a more uniform way verbs declare their inputs and outputs.